source/bao_hyp/config.rst

```diff
@@ -65,41 +65,512 @@ describe the image running on the VM (see `Guest Image`_);
+
+Using these macros not only streamlines the configuration steps but also ensures adherence to the
+correct syntax and parameters. Attempting to manually configure image details without utilizing
+these macros may result in errors or unintended behavior.
+
+The integration of the appropriate macro, tailored to your specific use case, is crucial for
+ensuring consistency and reliability in your VM setup. This macro should encompass essential
+parameters such as ``img_name``, ``base_addr``, ``load_addr``, and size. For instance:
+
+- ``IMAGE_BUILTIN``: Simplifies system configuration by leveraging Bao's default knowledge of the
+  image location. No separate configuration or loading of guest images through a bootloader is
+  required, and adjustments to the size of guest images are unnecessary.
+
+- ``IMAGE_LOADED``: Highly recommended, especially for MPU systems, where manual allocation of
+  space for the guest image can be challenging if embedded in Bao's binary. Without utilizing
+  LOADED, Bao may need to copy the image, potentially resulting in wasted space.
+
+Moreover, if the ``separately_loaded`` parameter is configured as false, the hypervisor interprets
+this setting as the offset of the built-in guest image within its own image, denoted as
+``VM_IMAGE_OFFSET``. During run-time, the hypervisor adjusts this value to be subsequently
+interpreted as a physical address. This adjustment involves adding the address at which the
+hypervisor itself was loaded. However, if the ``separately_loaded`` parameter is configured as
+true, the guest image is not embedded in the hypervisor image; instead, it is loaded independently.
+For more details, refer to the figure below.
+
+.. figure:: img/guest-image.svg
+    :align: center
+    :width: 100%
+    :name: vm-image-fig
+
+|
 
 Virtual Machine Configuration
 *****************************
 
+The VM configuration enables users to define the characteristics of each virtualized platform. It
+capabilities. The virtual machine configuration is performed by populating the structure ``struct
+vm_platform``, outlined below:
+
+.. code-block:: c
+
+    struct vm_platform {
+        size_t cpu_num;
+        size_t region_num;
+        struct vm_mem_region* regions;
+        size_t ipc_num;
+        struct ipc* ipcs;
+        size_t dev_num;
+        struct vm_dev_region* devs;
+        bool mmu;
+        struct arch_vm_platform arch;
+    }
+
+By customizing this configuration, users can tailor the virtual platform to suit specific workload
+requirements and application needs for their virtual machines. The configuration includes the
+definition of:
+
+- ``Number of CPUs`` - see details in `Number of vCPUs`_;
+- ``Memory regions`` - see details in `Memory Regions`_;`
+- ``Inter-Partition Comunication (IPC)`` - see details in `Inter-Partition Communication (IPC)`_;
+- ``Devices`` - see details in `Devices`_;
+- ``Architectural-Specific Configurations`` - see details in `Architectural-Specific
```

Comment:
```suggestion
- **Number of CPUs** - see details in `Number of vCPUs`_;
- **Memory regions** - see details in `Memory Regions`_;
- **Inter-Partition Comunication (IPC)** - see details in `Inter-Partition Communication (IPC)`_;
- **Devices** - see details in `Devices`_;
- **Architectural-Specific Configurations** - see details in `Architectural-Specific
```

Reply:
Thanks! Fixed in https://github.com/bao-project/bao-docs/pull/78/commits/9ffd91e472802b0f7d18b6ea36be3344f52e904f!